nixos/modules/services/security/openbao.nix

```diff
@@ -0,0 +1,274 @@
+{
+  config,
+  lib,
+  options,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.openbao;
+  opt = options.services.openbao;
+
+  configFile = pkgs.writeText "openbao.hcl" ''
+    # openbao in dev mode will refuse to start if its configuration sets listener
+    ${lib.optionalString (!cfg.dev) ''
+      listener "tcp" {
+        address = "${cfg.address}"
+        ${
+          if (cfg.tlsCertFile == null || cfg.tlsKeyFile == null) then
+            ''
+              tls_disable = "true"
```

Comment:
Please require some new option like `cfg.insecure` to enable this. If neither insecure nor (tlsCertFile && tlsKeyFile) are set, then it should be an assertion failure. The openbao documentation explicitly states that it defaults to https and recommends against http, so it is jarring to discover that we reverse that here.

Reply:
I think upstream concern is that you shouldn't provide a http only service. But that doesn't mean openbao needs to provide the TLS connection. You can have nginx configured as a reverse proxy also. Does this not address the security concerns?

Reply:
It's certainly true that there's other ways to achieve a secure connection, but the point is that the person who configures it should be making an explicit decision to have it listen insecurely. Upstream have deliberately chosen to make the system "secure by default, but you can opt for insecurity if if you want to go to the effort of designing an alternative system architecture", and this code changes it to "insecure by default, but you can opt for security". With this design, it's secretly necessary to think about all these other parts - a reverse proxy with tls enabled, making sure your listen address is ::1 or 127.0.0.1 or using a unix socket. So it doesn't address the concerns.